main .md

```diff
@@ -0,0 +1,396 @@
+ran = RandomForestRegressor(n_estimators = 50)
+ran.fit(x_train,y_train)
+ran.score(x_test, y_test)
+'''
+
+from joblib import Parallel, delayed
+import joblib
+
+# Save the model as a pickle in a file
+#joblib.dump(grad, 'mini model.pkl')
+
+# Load the model from the file
+l_joblib = joblib.load('mini model.pkl')
+print(l_joblib)
+
+
+                                    #prediction using Test data
+
+test_data=pd.read_csv("F:\Scientist\MinI project\Test.csv")
+print(test_data)
+ #Encoding categorical to numerical
+# Importing the required libary
+
+from sklearn import preprocessing
+
+print(test_data.head())
+print(test_data.columns)
+
+# Replacing LF with Low Fat, reg Regular, low fat with Low Fat
+test_data['Item_Fat_Content']= test_data['Item_Fat_Content'].replace({'LF':'Low Fat','reg':'Regular','low fat':'Low Fat'})
+print(f"--------------------------------------------------------",test_data['Item_Fat_Content'].value_counts())
+
+
+O_TYPE1 = pd.get_dummies(test_data['Outlet_Type'])
+O_LOCATION_T1 = pd.get_dummies(test_data['Outlet_Location_Type'])
+O_SIZE1 = pd.get_dummies(test_data['Outlet_Size'])
+I_FAT1 = pd.get_dummies(test_data['Item_Fat_Content'])
+
+df1 = pd.concat([test_data,O_TYPE1,O_LOCATION_T1,O_SIZE1,I_FAT1],axis=1)
+test = df1.drop(columns=['Item_Identifier','Item_Weight','Outlet_Identifier','Item_Type','Outlet_Establishment_Year',
+                         'Outlet_Size','Outlet_Type','Outlet_Location_Type','Item_Fat_Content'])
+print(test)
+#removing duplicated data
+print(sum(test_data.duplicated()))
+
+#again checking null values
+print(test_data.isnull().sum())
+
+duplicate=test_data.duplicated()
+print(duplicate.sum())
+
+test_data['Item_Weight']=test_data['Item_Weight'].fillna(value=test_data['Item_Weight'].mean())
+print(test_data['Item_Weight'])
+print(test_data['Item_Weight'].isnull().sum())
+
+#outlet size
+test_data['Outlet_Size']=test_data['Outlet_Size'].fillna('medium')
+
+print(test_data['Outlet_Size'])
+print(test_data['Outlet_Size'].isnull().sum())
+
+
+
+
+
+
+                                  #modling
+
+#importing required Ml library
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LinearRegression
+from sklearn.ensemble import GradientBoostingRegressor
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.linear_model import Lasso
+
+#tranfer data sets in training and testing
+x=train.iloc[:,:-1].values
+y=train.iloc[:,1].values
+print(x)
+print(y)
+
+x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.60,random_state=133)
+print( x_train.shape,y_train.shape)
+print( x_test.shape,y_test.shape)
+
+
+#train the model
+
+reqr=linear_model.LinearRegression()
+reqr.fit(x_train,y_train)
+
+scor=reqr.score(x_test,y_test)
+print(f"\t linear regression prediction\n",scor)
+
+
+#lassso re
+LS=Lasso(alpha=0.05)
+LS.fit(x_train,y_train)
+v=LS.score(x_test,y_test)
+print(v,f"\t lassso regression prediction\n")
+ #gradientBoostingRegressor
```

Comment:
#gradientBoostingRegressor

from xgboost import XGBClassifier
seed=7
text_size=0.33
x1 = train.iloc[:,0:8]
y2 = train.iloc[:,8]
print(x1)
print(y2)

x_train,x_test,y_train,y_test=train_test_split(x1,y2,test_size=text_size,random_state=seed)

model=XGBClassifier()
model.fit(x_train,y_train)
print(model)

y_predd=model.predict(x_test)
predicton=[round(value) for value in y_predd]
print(predicton)

accary=accuracy_score(y_test, predicton)
print("Accuracy: %.2f%%" % (accary * 100.0))

Reply:
using it in place of previous code